src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js

```diff
@@ -56,6 +56,11 @@
      */
     var read_config = function () {
         var cfg = default_cfg;
+
+        if (!liveNotebook) {
+            return cfg;
+        }
+
```

Comment:
This is not desirable. The idea (I am the author) was to be able to read the configuration stored in the notebook (line 66): the toc can be floating, the contents name may have been changed, etc

Reply:
at line 66, the variable `IPython` referenced before assignment ([catch require do nothing with variable](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/7144c90048a33197755503064ab1a2a496480b11/src/jupyter_contrib_nbextensions/nbextensions/toc2/toc2.js#L84-L100))

Thus, the idea, read some config, is meaningless while in no-live-notebook